website/docs/plugin/framework/getting-started/code-walkthrough.mdx

```diff
@@ -0,0 +1,381 @@
+- `Schema`: This function returns a data source `schema.Schema` struct that defines the data source schema. The schema specifies the constraints of the data source Terraform configuration block. It defines what fields a data source configuration block has, and gives Terraform metadata about those fields. For instance, defining whether a field is optional.
+- [`Read`]((/plugin/framework/data-sources#read-method)): This function lets the provider read data source values in order to update state.
+
+```go
+package provider
+
+import (
+    "context"
+
+    "github.com/hashicorp/terraform-plugin-framework/datasource"
+    "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+    "github.com/hashicorp/terraform-plugin-framework/types"
+    "github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var _ datasource.DataSource = (*exampleDataSource)(nil)
+
+type exampleDataSource struct {
+    provider exampleProvider
+}
+
+func NewDataSource() datasource.DataSource {
+    return &exampleDataSource{}
+}
+
+func (e *exampleDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+    resp.TypeName = req.ProviderTypeName + "_datasource"
+}
+
+func (e *exampleDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+    resp.Schema = schema.Schema{
+        Attributes: map[string]schema.Attribute{
+            "configurable_attribute": schema.StringAttribute{
+                MarkdownDescription: "Example configurable attribute",
+                Optional:            true,
+            },
+            "id": schema.StringAttribute{
+                MarkdownDescription: "Example identifier",
+                Computed:            true,
+            },
+        },
+    }
+}
+
+type exampleDataSourceData struct {
+    ConfigurableAttribute types.String `tfsdk:"configurable_attribute"`
+    Id                    types.String `tfsdk:"id"`
+}
+
+func (e *exampleDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+    var data exampleDataSourceData
+
+    diags := req.Config.Get(ctx, &data)
+    resp.Diagnostics.Append(diags...)
+
+    if resp.Diagnostics.HasError() {
+        return
+    }
+
+	// Interact with 3rd party API to read data source.
+
+    data.Id = types.StringValue("example-id")
+
+    tflog.Trace(ctx, "read a data source")
+
+    diags = resp.State.Set(ctx, &data)
+    resp.Diagnostics.Append(diags...)
+}
+```
+
+Refer to [Data Sources](/plugin/framework/data-sources) for more details and configuration examples.
+
+## Terraform Configuration
+
+With the definitions we have for [provider server](#provider-server), [provider](#provider), [resource](#resource) and [data source](#data-source), we can run the provider by specifying configuration and executing `terraform apply`.
```

Comment:
Not for right now, but I'm a little worried this may give the impression that the provider developer can just create a `*.tf` file with the below and it'll magically work. Not sure if it is worth adding words or pointing to other documentation to mention how to wire/publish things together fully.